src/app/core/components/search/search.component.ts

```diff
@@ -27,7 +28,7 @@ export class SearchComponent {
 		desc: ['']
 	});
 
-	constructor(private courseService: CourseService, private sanitizer: DomSanitizer, public fb: FormBuilder) {
+	constructor(private courseService: CourseService, private sanitizer: DomSanitizer, public fb: FormBuilder, private filter: FilterPipe) {
```

Comment:
It would be easier to read if you list parameters in column.
```typescript
constructor(
    private courseService: CourseService,
    private sanitizer: DomSanitizer, 
    public fb: FormBuilder, 
    private filter: FilterPipe
) {
```